src/components/pages/ServiceMarketplace/index.tsx

```diff
@@ -50,27 +50,30 @@ import {
 import SortImage from 'components/shared/frame/SortImage'
 import { ServiceTypeIdsEnum } from 'features/serviceManagement/apiSlice'
 import NoItems from '../NoItems'
+import { serviceTypeMapping } from 'types/Constants'
 
 dayjs.extend(isToday)
 dayjs.extend(isYesterday)
 dayjs.extend(relativeTime)
 
+const serviceTypeIdMapping: Record<string, ServiceTypeIdsEnum> = {
+  [ServiceTypeIdsEnum.DATASPACE_SERVICES]: ServiceTypeIdsEnum.DATASPACE_SERVICE,
+  [ServiceTypeIdsEnum.CONSULTANCY_SERVICES]:
+    ServiceTypeIdsEnum.CONSULTANCY_SERVICE,
+}
+
 export default function ServiceMarketplace() {
   const { t } = useTranslation()
   const theme = useTheme()
   const [searchExpr, setSearchExpr] = useState<string>('')
   const [showModal, setShowModal] = useState<boolean>(false)
-  const [selected, setSelected] = useState<string>('All Services')
+  const [selected, setSelected] = useState<string>(
+    t('content.serviceMarketplace.tabs.all')
+  )
   const [sortOption, setSortOption] = useState<string>('new')
   const [cardServices, setCardServices] = useState<ServiceRequest[]>([])
 
-  let serviceTypeId = ''
-
-  if (selected === ServiceTypeIdsEnum.DATASPACE_SERVICES) {
-    serviceTypeId = ServiceTypeIdsEnum.DATASPACE_SERVICE
-  } else if (selected === ServiceTypeIdsEnum.CONSULTANCY_SERVICES) {
-    serviceTypeId = ServiceTypeIdsEnum.CONSULTANCY_SERVICE
-  }
+  const serviceTypeId = serviceTypeIdMapping[serviceTypeMapping[selected]] ?? ''
```

Comment:
empty string -> undefined

See my comments here: https://github.com/eclipse-tractusx/portal-frontend/pull/1134#pullrequestreview-2369652213

Reply:
sorry I didn't understand what you mean by "**this would be affected by the change from empty string to undefined**" which thing you are referring to please ? though I have replaced empty string with undefined.

Reply:
@oyo  I have removed the empty string from   `const serviceTypeId = serviceTypeIdMapping[serviceTypeMapping[selected]] ?? ''` . However, backend expects empty string incase of no selection so I added empty string in query for API here `serviceType: serviceTypeId ?? ''`  as per previous request payload.

Reply:
Hi @Usmanfee I got feedback from @oyo, please see https://github.com/eclipse-tractusx/portal-frontend/pull/1209#discussion_r1832386769